CMake/external_libcurl.cmake

```diff
@@ -11,7 +11,8 @@ if(CHECK_FOR_UPDATES)
     if (WIN32)
         set(CURL_FLAGS ${CURL_FLAGS} -DCMAKE_USE_SCHANNEL=ON )
     else()
-        set(CURL_FLAGS ${CURL_FLAGS} -DCMAKE_USE_OPENSSL=ON )
+        find_package(OpenSSL REQUIRED)
+        set(CURL_FLAGS ${CURL_FLAGS} -DCMAKE_USE_OPENSSL=ON -DOPENSSL_ROOT_DIR=${OPENSSL_ROOT_DIR})
```

Comment:
find_package(OpenSSL REQUIRED) does not set "OPENSSL_ROOT_DIR",
Did you set "OPENSSL_ROOT_DIR"  env var before the build when you tested it?

When I run it on Linux/OSX without setting it then I get ${OPENSSL_ROOT_DIR} = ""

Try to see if [this](https://github.com/IntelRealSense/librealsense/blob/d5e775b75f7adbf75c911b7c85a4afd2f16defcb/.travis.yml#L98) works for you

Reply:
Ah, I moved the `find_package(OpenSSL REQUIRED)` up because I assumed it would set `OPENSSL_ROOT_DIR` if it could find it. [It looks like it doesn't though](https://cmake.org/cmake/help/latest/module/FindOpenSSL.html) so that might be unnecessary. I can change that back.

Setting the environment variable before the cmake/make commands like
```
export OPENSSL_ROOT_DIR=`brew --prefix openssl`
cmake ..
make libcurl
```
seems to build successfully. This was meant to fix the case where you set it in cmake or via the GUI like `cmake .. -DOPENSSL_ROOT_DIR=/path/to/openssl`

Does this affect the linux build if the variable is not set? I can't access the CI website right now but I thought I saw that Linux builds were still working.

Reply:
at a minimum I think the docs should be updated with either method though, it seems like other macos users stumble on this as well

Reply:
At Linux we didn't had to set the "OPENSSL_ROOT_DIR" as it was installed at a known location.
But it's user dependent.
You are right about the required update of the documentation on Linux / MacOs installation, we will surely update it.
I'm glad you popped up the issue to trigger it.

Keep it up :)
We are always here to help.
Thanks and good luck

Reply:
yeah I think one possible reason why `find_package(OpenSSL)` doesn't work on macOS is that homebrew doesn't symlink it to avoid conflicting with LibreSSL
```
$ brew reinstall openssl@1.1
...
openssl@1.1 is keg-only, which means it was not symlinked into /usr/local,
because macOS provides LibreSSL.
```